src/Builder/Visitor/AddPluginToPluginListVisitor.php

```diff
@@ -320,9 +320,12 @@ protected function isPluginAdded(Array_ $node): bool
             if ($item === null || !($item->value instanceof New_)) {
                 continue;
             }
+
             $nodeClassName = $item->value->class->toString();
+            $usedParentClasses = class_exists($nodeClassName) ? (class_parents($nodeClassName) ?: []) : [];
+            $classToAdd = $this->classMetadataTransfer->getSourceOrFail();
```

Comment:
I suggest moving the line before `foreach` 
And to add a new line in `if` section 

But these are not critical stuff 

` if ($this->isKeyEqualsToCurrentOne($item) &&
                ($nodeClassName === $classToAdd || in_array($classToAdd, $usedParentClasses)))
            {
                return true;
            }`